### PR TITLE
 Add human-ethers, add to transfer error

### LINKED ISSUE #169: Improve error presentation to user.
image

```
user rejected action (action="sendTransaction", reason="rejected", info={ "error": { "code": 4001, "data": { "request": { "method": "eth_sendTransaction", "params": [ { "data": "0xa9059cbb000000000000000000000000d8da6bf26964af9d7eed9e03e53415d37aa9604500000000000000000000000000000000000000000000000000000f478e084000", "from": "0xd8da6bf26964af9d7eed9e03e53415d37aa96045", "gas": "0x8ee2", "to": "0x1f9840a85d5af5bf1d1762f925bdaddc4201f984" } ] } }, "message": "ethers-user-denied: Interceptor Tx Signature: User denied transaction signature." }, "payload": { "id": 52, "jsonrpc": "2.0", "method": "eth_sendTransaction", "params": [ { "data": "0xa9059cbb000000000000000000000000d8da6bf26964af9d7eed9e03e53415d37aa9604500000000000000000000000000000000000000000000000000000f478e084000", "from": "0xd8da6bf26964af9d7eed9e03e53415d37aa96045", "gas": "0x8ee2", "to": "0x1f9840a85d5af5bf1d1762f925bdaddc4201f984" } ] } }, code=ACTION_REJECTED, version=6.6.1)
```

We should parse the error and look for and handle well known error codes.  We should work with @KillariDev to ensure we at least support all of the Interceptor errors and present them to the user in a nice clean way.

We should still include the JSON, but perhaps hidden behind an Additional Details <details> section or something.

Note: this is relatively low priority.

### PR DESCRIPTION
**Pull request by 0xjimmy**
Fixes #169

Upgrades Ethers to 6.7.1
Adds human-readable-ethers-error.ts

@jubalm, you may want to style the error, I just display the plain text human error.

### CODE REVIEW

**Review by KillariDev — APPROVED** [review]
looks good, but lets wait for Jubals look on it before merging

File: app/ts/library/human-ethers-errors.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { ActionRejectedError, NetworkError, TimeoutError, CancelledError, NumericFaultError, MissingArgumentError, CallExceptionError, TransactionReplacedError, ReplacementUnderpricedError, UnconfiguredNameError, UnexpectedArgumentError, InvalidArgumentError, BufferOverrunError, BadDataError, ServerError, NotImplementedError, UnknownError, UnsupportedOperationError, InsufficientFundsError, NonceExpiredError, OffchainFaultError } from 'ethers'
+
+type ETHERS_ERROR = (UnknownError & { code: 'UNKNOWN_ERROR' })
```
Comment (jubalm) [inline app/ts/library/human-ethers-errors.ts:3]:
```suggestion
type EthersError = (UnknownError & { code: 'UNKNOWN_ERROR' })
```

File: app/ts/library/human-ethers-errors.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  | (NonceExpiredError & { code: 'NONCE_EXPIRED' })
+  | (OffchainFaultError & { code: 'OFFCHAIN_FAULT' })
+  | (ReplacementUnderpricedError & { code: 'REPLACEMENT_UNDERPRICED' })
+  | (TransactionReplacedError & { code: 'TRANSACTION_REPLACED' })
+  | (UnconfiguredNameError & { code: 'UNCONFIGURED_NAME' })
+  | (ActionRejectedError & { code: 'ACTION_REJECTED' })
+const ETHERS_ERROR_CODES = [ 'UNKNOWN_ERROR', 'NOT_IMPLEMENTED', 'UNSUPPORTED_OPERATION', 'NETWORK_ERROR', 'SERVER_ERROR', 'TIMEOUT', 'BAD_DATA', 'CANCELLED', 'BUFFER_OVERRUN', 'NUMERIC_FAULT', 'INVALID_ARGUMENT', 'MISSING_ARGUMENT', 'UNEXPECTED_ARGUMENT', 'CALL_EXCEPTION', 'INSUFFICIENT_FUNDS', 'NONCE_EXPIRED', 'OFFCHAIN_FAULT', 'REPLACEMENT_UNDERPRICED', 'TRANSACTION_REPLACED', 'UNCONFIGURED_NAME', 'ACTION_REJECTED' ] as const
+type ETHERS_ERROR_CODES = typeof ETHERS_ERROR_CODES[number]
+
+export function isEthersError(error: unknown): error is ETHERS_ERROR {
```
Comment (jubalm) [inline app/ts/library/human-ethers-errors.ts:27]:
```suggestion
export function isEthersError(error: unknown): error is EthersError {
```

**Review by jubalm — APPROVED** [review]
rebased this branch as there are already changes on top of it from a recently merged PR, adopting the library. moving the error handlers  on #210

**Pull request merged**